core-bundle/src/Security/Voter/BackendAccessVoter.php

```diff
@@ -28,10 +38,50 @@ protected function voteOnAttribute($attribute, $subject, TokenInterface $token):
         $user = $token->getUser();
         [, $field] = explode('.', $attribute, 2);
 
-        if (!$user instanceof BackendUser || (!is_scalar($subject) && !\is_array($subject))) {
+        if (!$user instanceof BackendUser) {
+            return false;
+        }
+
+        if ('can_edit_fields' === $field) {
+            return $this->canEditFieldsOf($subject, $user);
+        }
+
+        if (isset(self::ALLOWED_FLAGS[$field])) {
+            return $this->isAllowed($subject, self::ALLOWED_FLAGS[$field], $user);
+        }
+
+        return $this->hasAccess($subject, $field, $user);
+    }
+
+    private function hasAccess($subject, string $field, BackendUser $user): bool
+    {
+        if (!is_scalar($subject) && !\is_array($subject)) {
             return false;
         }
 
         return $user->hasAccess($subject, $field);
     }
+
+    private function isAllowed($subject, int $flag, BackendUser $user): bool
+    {
+        if ($subject instanceof PageModel) {
```

Comment:
What if `$subject` is another model? Then the `is_array()` check below will fail and `false` will be returned, so effectively, `PageModel` is the only supported non-array subject. Intended?

Reply:
Yes, that is intended. The `BackendUser::isAllowed` method expects an array of the current page data. For convenience, I added the option to pass a page model.